resources/migrations/001_update_migrations.yaml

```diff
@@ -8400,6 +8400,46 @@ databaseChangeLog:
                   constraints:
                     nullable: false
 
+  - changeSet:
+      id: v51.2024-07-08T10:00:01
+      author: qnkhuat
+      comment: Create pulse_channel.channel_id
+      preConditions:
+        - onFail: MARK_RAN
+        - not:
+            - columnExists:
+                tableName: pulse_channel
+                columnName: channel_id
+      changes:
+        - addColumn:
+            tableName: pulse_channel
+            columns:
+              - column:
+                  name: channel_id
+                  type: integer
+                  remarks: The channel ID
+
+  - changeSet:
+      id: v51.2024-07-08T10:00:02
+      author: qnkhuat
+      comment: Add fk constraint to pulse_channel.channel_id
+      preConditions:
+        - onFail: MARK_RAN
```

Comment:
I'm worried about just skipping the constraint, as we won't have cascading deletes in this case. Can you explain why this is OK, I feel like I am missing something.

Reply:
I added this because we have a linter to suggest adding preconditions to migrations idempotence see https://github.com/metabase/metabase/blob/dev-tool-monitor-toucan2-queries/bin/lint-migrations-file/src/change_set/strict.clj#L135

I don't think it's possible to hit this though, unless someone create this exact fk constraint manually, in which case I trust that they know what they're doing :)

Reply:
I would think that prior condition would prevent that failure though, this would have to be bad references I think.

Reply:
that onfail is applied to the precondition only, not the whole migration.
it means if there is an FK constraint existing, then mark the migration as “RAN” or else attempt to create the FK constraint.

if it fails to create the FK for whatever reasons, then migration will fail and stop like normal